sklearn/utils/validation.py

```diff
@@ -2629,13 +2636,14 @@ def _check_pos_label_consistency(pos_label, y_true):
     # when elements in the two arrays are not comparable.
     if pos_label is None:
         # Compute classes only if pos_label is not specified:
-        classes = np.unique(y_true)
-        if classes.dtype.kind in "OUS" or not (
-            np.array_equal(classes, [0, 1])
-            or np.array_equal(classes, [-1, 1])
-            or np.array_equal(classes, [0])
-            or np.array_equal(classes, [-1])
-            or np.array_equal(classes, [1])
+        xp, _, device = get_namespace_and_device(y_true)
+        classes = xp.unique_values(y_true)
+        if (_is_numpy_namespace(xp) and classes.dtype.kind in "OUS") or not (
+            xp.all(classes == xp.asarray([0, 1], device=device))
+            or xp.all(classes == xp.asarray([-1, 1], device=device))
+            or xp.all(classes == xp.asarray([0], device=device))
+            or xp.all(classes == xp.asarray([-1], device=device))
+            or xp.all(classes == xp.asarray([1], device=device))
         ):
             classes_repr = ", ".join([repr(c) for c in classes.tolist()])
```

Comment:
Do you know why `.tolist()`  works here? I don't think it is part of the array API standard and `classes` is a array API array, which is why I am surprised that this works. Are we just getting lucky or does it make sense that this works?

Reply:
I'd convert `classes` to a numpy array and then use it here. It should be short, so cheap to convert with `_convert_to_numpy` from `sklearn.utils._array_api`